lang/LangPrimSource/SC_Ableton_Link.cpp

```diff
@@ -37,30 +22,39 @@ class LinkClock : public TempoClock
 	void SetTempoAtBeat(double inTempo, double inBeats);
 	void SetTempoAtTime(double inTempo, double inSeconds);
 	void SetAll(double inTempo, double inBeats, double inSeconds);
-	double BeatsToSecs(double beats) const
-	{
-		auto sessionState = mLink.captureAppSessionState();
-		double secs = linkToHrTime(sessionState.timeAtBeat(beats, mQuantum)) - mLatency;
-		return secs;
-	}
-	double SecsToBeats(double secs) const
-	{
-		auto sessionState = mLink.captureAppSessionState();
-		double beats = sessionState.beatAtTime(hrToLinkTime(secs + mLatency), mQuantum);
-		return beats;
-	}
+	double BeatsToSecs(double beats) const;
+	double SecsToBeats(double secs) const;
 
 	void SetQuantum(double quantum);
         double GetLatency();
         void SetLatency(double latency);
 	std::size_t NumPeers() const { return mLink.numPeers(); }
 
+  static void Init();
+  static std::chrono::microseconds GetInitTime() { return LinkClock::mInitTime; }
+
 private:
 	ableton::Link mLink;
 	double mQuantum;
 	double mLatency;
+  static std::chrono::microseconds mInitTime;
 };
 
+std::chrono::microseconds LinkClock::mInitTime;
+
+inline std::chrono::microseconds hrToLinkTime(double secs){
+	auto time = std::chrono::duration<double>(secs);
+	return std::chrono::duration_cast<std::chrono::microseconds>(time) + LinkClock::GetInitTime();
+}
+
+inline double linkToHrTime(std::chrono::microseconds micros){
+	return DurToFloat(micros - LinkClock::GetInitTime());
+}
```

Comment:
Helper methods `hrToLinkTime` and `linkToHrTime` need to move below `LinkClock` because they depend on it now.